scripts/_RosTeamWs_Defines.bash

```diff
@@ -237,69 +279,111 @@ function print_and_exit {
-  fi
-  RosTeamWS_FRAMEWORK_SCRIPTS_PATH="$FRAMEWORK_PACKAGE_PATH/scripts/"
+  echo -e "${TERMINAL_COLOR_USER_INPUT_DECISION}${question}[${rtw_accepted_answers[*]}]${TERMINAL_COLOR_NC}"
+  read user_answer
+
+  while ! $(is_accepted_user_answer "$user_answer");
+  do
+    echo -e "${TERMINAL_COLOR_USER_INPUT_DECISION}${question} Please type one of the following: [${rtw_accepted_answers[*]}]${TERMINAL_COLOR_NC}"
+    read user_answer
+  done
+}
+
+function set_framework_default_paths {
+  FRAMEWORK_NAME="ros_team_workspace"
+  FRAMEWORK_BASE_PATH="$(RosTeamWS_script_own_dir)/.."
 
+  RosTeamWS_FRAMEWORK_SCRIPTS_PATH="$FRAMEWORK_BASE_PATH/scripts"
+  RosTeamWS_FRAMEWORK_OS_CONFIGURE_PATH="$RosTeamWS_FRAMEWORK_SCRIPTS_PATH/os_configure"
   # Script-specific variables
-  PACKAGE_TEMPLATES="$FRAMEWORK_PACKAGE_PATH/templates/package"
-  ROBOT_DESCRIPTION_TEMPLATES="$FRAMEWORK_PACKAGE_PATH/templates/robot_description"
-  ROS2_CONTROL_TEMPLATES="$FRAMEWORK_PACKAGE_PATH/templates/ros2_control"
+  PACKAGE_TEMPLATES="$FRAMEWORK_BASE_PATH/templates/package"
+  ROBOT_DESCRIPTION_TEMPLATES="$FRAMEWORK_BASE_PATH/templates/robot_description"
+  ROS2_CONTROL_TEMPLATES="$FRAMEWORK_BASE_PATH/templates/ros2_control"
   ROS2_CONTROL_HW_ITF_TEMPLATES="$ROS2_CONTROL_TEMPLATES/hardware"
   ROS2_CONTROL_CONTROLLER_TEMPLATES="$ROS2_CONTROL_TEMPLATES/controller"
-  LICENSE_TEMPLATES="$FRAMEWORK_PACKAGE_PATH/templates/licenses"
-  DOCKER_TEMPLATES="$FRAMEWORK_PACKAGE_PATH/templates/docker"
+  LICENSE_TEMPLATES="$FRAMEWORK_BASE_PATH/templates/licenses"
+  DOCKER_TEMPLATES="$FRAMEWORK_BASE_PATH/templates/docker"
+}
+
+function is_valid_ros_distribution {
+    local ros_distribution=$1
+    declare -a supported_ros_versions=("${!2}")
+
+  if [[ " ${supported_ros_versions[*]} " =~ " ${ros_distribution} " ]]; then
+    return
+  fi
+
+  false
 }
 
 function check_ros_distro {
   ros_distro=$1
-  if [ -z "$1" ]; then
-    if [ -n "$DEFAULT_ROS_DISTRO" ]; then
-      ros_distro=$DEFAULT_ROS_DISTRO
-      echo "No ros_distro defined. Using default: '$ros_distro'"
-      if [ ! -d "/opt/ros/$ros_distro" ]; then
-        print_and_exit "FATAL: ROS '$ros_distro' not installed on this computer! Exiting..."
-  #             echo "FATAL: ROS '$ros_distro' not installed on this computer! Exiting..."
-  #             exit
-      fi
-      read -p "Press <ENTER> to continue or <CTRL>+C to exit."
+
+  # check if the given distribution is a distribution supported by rtw
+  while ! is_valid_ros_distribution "$ros_distro" rtw_supported_ros_distributions[@];
+  do
+      echo -e "${TERMINAL_COLOR_USER_INPUT_DECISION}The ros distribution {${ros_distro}} you chose is not supported by RosTeamWS. Please chose either of the following:${rtw_supported_ros_distributions[*]}"
+      read ros_distro
+  done
+
+  if [ ! -d "/opt/ros/$ros_distro" ]; then
+    local upper_case=$(echo $ros_distro | tr '[:lower:]' '[:upper:]')
+    local alternative_ros_location=ALTERNATIVE_ROS_${upper_case}_LOCATION
```

Comment:
Why do we need an alternative ROS location? We are actually interested in alternative RosTeamWorkspace locations. Can you give me a short example when this functionality is needed?

Reply:
If someone has installed ros at a different location than `/opt/ros/` (e.g. source install) we check if the requested ros distro is installed there. Otherwise, we cannot create a new package for example since this depends on `ros2 pkg create`.

Reply:
If make rtw independent from ros in the next step it could be possible that we don't need this anymore.

Reply:
OK, we can leave it, this is good!